### PR TITLE
Implementation of KTrajectoryRadial2D

### LINKED ISSUE #11: 2D radial trajectory
Create 2D (Golden) radial trajectory

Check list:
- [x]     Create branch
- [x]     Implement code
- [ ]     Add comments
- [ ]     Implement tests
- [ ]     Squash commmits
- [ ]     Create PR
- [ ]     Assign reviewer(s) to PR

### PR DESCRIPTION
**Pull request by schote**
`KTrajectoryRadial2D`
- Code implementation
- Documentation
- Tests

closes #11

### CODE REVIEW

File: tests/data/test_traj_calculators.py [hunk 1]
```diff
@@ -18,7 +18,49 @@
...
+    
+    return random_kheader
+
+def radial2D_traj_shape(valid_rad2d_kheader):
+    """Expected shape of trajectory based on KHeader."""
+    nk0 = valid_rad2d_kheader.acq_info.number_of_samples[0, 0, 0]
+    nk1 = valid_rad2d_kheader.acq_info.idx.k1.shape[2]
+    nk2 = valid_rad2d_kheader.acq_info.idx.k1.shape[1]
+    nother = 1
+    return (torch.Size([nother, nk2, nk1, 1]), torch.Size([nother, nk2, nk1, 1]), torch.Size([nother, 1, 1, nk0]))
```
Comment (fzimmermann89) [inline tests/data/test_traj_calculators.py:54]: this is wrong imho

File: src/mrpro/data/traj_calculators/_KTrajectoryRadial2D.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        # K-space locations along phase encoding lines
+        krad = self._krad(kheader)
+        
+        # Angles of phase encoding lines
+        kang = self._kang(kheader) 
+        
+        # K-space cartesian coordinates
+        kx = krad[None, None, None] * torch.cos(kang)[..., None]
+        ky = krad[None, None, None] * torch.sin(kang)[..., None]
+        kz = torch.zeros(kx.shape)
```
Comment (fzimmermann89) [inline src/mrpro/data/traj_calculators/_KTrajectoryRadial2D.py]: kz=torch.zeros(1,1,1,1)

File: src/mrpro/data/traj_calculators/_KTrajectoryRadial2D.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+from mrpro.data.traj_calculators import KTrajectoryCalculator
+
+
+class KTrajectoryRadial2D(KTrajectoryCalculator):
+    """Radial 2D trajectory.
+
+    Parameters
+    ----------
+    angle
+        angle in rad between two radial phase encoding lines
```
Comment (ckolbPTB) [inline src/mrpro/data/traj_calculators/_KTrajectoryRadial2D.py]: should be **radial lines** not **radial phase encoding** lines

File: src/mrpro/data/traj_calculators/_KTrajectoryRadial2D.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    """
+
+    def __init__(
+        self,
+        angle: float = np.deg2rad(180 / ((1 + np.sqrt(5)) / 2)),
```
Comment (ckolbPTB) [inline src/mrpro/data/traj_calculators/_KTrajectoryRadial2D.py]: should be torch.pi * 0.618034

File: src/mrpro/data/traj_calculators/_KTrajectoryRadial2D.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        ----------
+        kheader
+           MR raw data header (KHeader) containing required meta data
+
+        Returns
+        -------
+            radial 2D trajectory for given KHeader
+        """
+
+        # K-space locations along phase encoding lines
```
Comment (ckolbPTB) [inline src/mrpro/data/traj_calculators/_KTrajectoryRadial2D.py]: K-space locations along readout

File: src/mrpro/data/traj_calculators/_KTrajectoryRadial2D.py [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+        krad = self._krad(kheader)
+
+        # Angles of phase encoding lines
```
Comment (ckolbPTB) [inline src/mrpro/data/traj_calculators/_KTrajectoryRadial2D.py]: Angles of readout lines

File: tests/data/test_traj_calculators.py [hunk 7]
```diff
@@ -16,9 +16,47 @@
...
+@pytest.fixture(scope='function')
+def valid_rad2d_kheader(monkeypatch, random_kheader):
+    """KHeader with all necessary parameters for radial 2D trajectories."""
+    # K-space dimensions
+    nk0 = 200
+    nk1 = 20
+
+    # List of k1 indices in the shape
+    idx_k1 = torch.arange(nk1, dtype=torch.int32)[None, None, ...]
+    # idx_k1 = k1[..., None].repeat(1, nk0)[None, ...]
```
Comment (ckolbPTB) [inline tests/data/test_traj_calculators.py]: remove commented line

File: tests/data/test_traj_calculators.py [hunk 8]
```diff
@@ -16,9 +16,47 @@
...
+
+    # Set parameters for Radial 2D trajectory
```
Comment (ckolbPTB) [inline tests/data/test_traj_calculators.py]: radial not **R**adial

**Review by ckolbPTB — COMMENTED** [review]
some small comments

**Review by ckolbPTB — APPROVED** [review]

**Pull request merged**